packages/react-core/package.json

```diff
@@ -19,13 +19,15 @@
   ],
   "scripts": {
     "clean": "rimraf dist node_modules",
-    "prebuild": "rimraf dist",
     "build": "rollup --config",
-    "build:esm": "tsc --project tsconfig.esm.json",
-    "dev": "yarn build:esm --watch",
-    "lint": "tsc --noEmit && eslint src",
-    "test": "jest --coverage --verbose",
-    "test:watch": "yarn test --watch"
+    "dev": "yarn build --watch",
+    "lint": "yarn typecheck && eslint src",
+    "prebuild": "rimraf dist",
+    "test": "yarn test:unit && yarn test:esm",
```

Comment:
doesn't need to be addressed here, but should we reverse these so we can pass file paths to jest?

Reply:
Think that's what `test:unit` is for, although i personally prefer `test` to just correspond with running jest. Switching the order of commands would allow appending a path, but that feels a bit fragile to me. Alternatively we could add a CI specific test script that runs the ESM check and the unit tests to decouple the two from `test`. Open to discussion 🙂

Reply:
I like the idea of putting it in a CI specific test script, since this isn't something that should need to run over and over again locally, and could primarily be caught in PRs

Reply:
Cool cool, so thinking that we would then have something like:
```
"check:esm": "node --input-type=module --eval 'import \"@aws-amplify/ui\"'",
"test": "jest",
"test:ci": "yarn test && yarn check:esm",
"test:watch": "yarn test --watch",
```

Make sense?

Reply:
I like the change you mentioned, let's just be careful we don't break CI with this because `yarn test` for unit tests: 

https://github.com/aws-amplify/amplify-ui/blob/1bf5b330d108fb4eee28f3951fd5b0a45b4547b0/.github/workflows/reusable-e2e.yml#L181

Reply:
I think it'd make sense to do a separate PR for introducing `test:ci`, so that we don't have to worry about changing workflow in this PR